classes/phing/types/selectors/SizeSelector.php

```diff
@@ -28,186 +28,60 @@
-        "Gi",
-        "GI",
-        "gi",
-        "gibi",
-        "GIBI",
-        "T",
-        "t",
-        "tera",
-        "TERA",
-        /* You wish! */
-        "Ti",
-        "TI",
-        "ti",
-        "tebi",
-        "TEBI"
-    ];
+    private const VALUE_KEY = 'value';
+    private const WHEN_KEY  = 'when';
+    private const WHEN      = [-1 => 'less',
+                               0  => 'equal',
+                               1  => 'more',];
 
-    /**
-     * @return string
-     */
-    public function __toString()
+    public function __toString(): string
     {
-        $buf = "{sizeselector value: ";
-        $buf .= $this->sizelimit;
-        $buf .= "compare: ";
-        if ($this->cmp === 0) {
-            $buf .= "less";
-        } elseif ($this->cmp === 1) {
-            $buf .= "more";
-        } else {
-            $buf .= "equal";
-        }
-        $buf .= "}";
-
-        return $buf;
+        $format = '{%s value: %s when: %s}';
+        return sprintf($format, __CLASS__, $this->value, $this->when);
     }
 
     /**
-     * A size selector needs to know what size to base its selecting on.
-     * This will be further modified by the multiplier to get an
-     * actual size limit.
+     * Filesize
      *
-     * @param int $size the size to select against expressed in units
+     * Possible values are: '1024', '5000B', '300M', '2G'.
      *
-     * @return void
-     */
-    public function setValue($size)
-    {
-        $this->size = $size;
-        if (($this->multiplier !== 0) && ($this->size > -1)) {
-            $this->sizelimit = $size * $this->multiplier;
-        }
-    }
-
-    /**
-     * Sets the units to use for the comparison. This is a little
-     * complicated because common usage has created standards that
-     * play havoc with capitalization rules. Thus, some people will
-     * use "K" for indicating 1000's, when the SI standard calls for
-     * "k". Others have tried to introduce "K" as a multiple of 1024,
-     * but that falls down when you reach "M", since "m" is already
-     * defined as 0.001.
-     * <p>
-     * To get around this complexity, a number of standards bodies
-     * have proposed the 2^10 standard, and at least one has adopted
-     * it. But we are still left with a populace that isn't clear on
-     * how capitalization should work.
-     * <p>
-     * We therefore ignore capitalization as much as possible.
-     * Completely mixed case is not possible, but all upper and lower
-     * forms are accepted for all long and short forms. Since we have
-     * no need to work with the 0.001 case, this practice works here.
-     * <p>
-     * This function translates all the long and short forms that a
-     * unit prefix can occur in and translates them into a single
-     * multiplier.
-     *
-     * @param array $units The units to compare the size to.
+     * @param string $value
      *
      * @return void
      */
-    public function setUnits($units)
```

Comment:
Please revert this function. Mix different things is not that good.
Why should a `value` know about its unit? This will make build scripts more complex (having properties twice one with and one without unit)